tensor2tensor/utils/registry.py

```diff
@@ -44,103 +44,33 @@ class MyModel(T2TModel):
 from __future__ import division
 from __future__ import print_function
 
+import tensorflow.contrib.framework as framework
+from tensorflow.python.util import tf_inspect as inspect
 import collections
 
 from tensor2tensor.utils import misc_utils
-import tensorflow as tf
-from tensorflow.python.util import tf_inspect as inspect
-
-_ATTACKS = {}
-_ATTACK_PARAMS = {}
-_HPARAMS = {}
-_MODELS = {}
-_PROBLEMS = {}
-_PRUNING_PARAMS = {}
-_PRUNING_STRATEGY = {}
-_RANGED_HPARAMS = {}
-
-# Key: registry name, Value: Registry
-_GENERIC_REGISTRIES = {}
-Registry = collections.namedtuple(
-    "_Registry", ["register", "get", "list", "registry"])
 
 
-def registry(registry_name):
-  """Returns `Registry` created by `create_registry`."""
-  if registry_name not in _GENERIC_REGISTRIES:
-    raise KeyError("No registry named %s. Available:\n%s" % (
-        registry_name, sorted(_GENERIC_REGISTRIES)))
-  return _GENERIC_REGISTRIES[registry_name]
+def default_name(class_or_fn):
+  """Default name for a class or fn.
 
-
-def create_registry(registry_name):
-  """Create a generic object registry.
+  This is the naming function by default for registers expecting classes or
```

Comment:
s/registers/registries/g

and in other places where you refer to register (as a noun) it should be registry